pkg/services/ngalert/schedule/schedule.go

```diff
@@ -225,23 +236,28 @@ func (sch *schedule) processTick(ctx context.Context, dispatcherGroup *errgroup.
 	missingFolder := make(map[string][]string)
 	for _, item := range alertRules {
 		key := item.GetKey()
-		ruleInfo, newRoutine := sch.registry.getOrCreateInfo(ctx, key)
+		if item.IsPaused {
```

Comment:
Should this be done in the database query that returns the schedulable rules?

Reply:
If you do it that way, how do you know what currently running alerts you have to pause?

Reply:
Is this because we cannot know if a running rule is deleted or paused?

Reply:
Yes

Reply:
Why do we care if it was paused or deleted from the perspective of the scheduler? In either case, don't we just stop running ti and then remove it from the scheduler?

Reply:
Why do we care if it was paused or deleted from the perspective of the scheduler? In either case, don't we just stop running it and then remove it from the scheduler?

Reply:
I understand, but why can't we do that from the API rather than the scheduler? Is it because we want to know the specific timestamp that evaluation was paused, instead of the timestamp that the database was updated?

Reply:
He could add that from the API. But don't you think that should be a scheduler responsibility? There is 2 ways of doing that:
1. Change the attribute from the API and change the value in DB. And, separately and asynchronously, make the schedule aware of those changes and act accordingly to that (current implementation).
2. Change the attribute from the API, change the value in DB, and make a call to the scheduler to trigger the change in alert state.

I feel option 2 is a bit intrusive in scheduler responsibilities.

Reply:
> why can't we do that from the API rather than the scheduler?

Unfortunately, this is not possible because the rule state is shared between scheduler (which holds "rule version" - part of the state) and state manager that holds the alert instances that are coupled with rule version. Therefore, all changes in state should be synchronously made via scheduler.

I have plans on moving the rule version to the state manager as part of work to process rules in group synchronously. After that, we can change the logic and let API and scheduler access state manager independently.